cycle2/analysis/wrangle.R

```diff
@@ -1,293 +1,247 @@
-  tools[[i]]<-ifelse(grepl(card10, toolChoices[[i]]) == TRUE, 10, tools[[i]])
-  tools[[i]]<-ifelse(grepl(card11, toolChoices[[i]]) == TRUE, 11, tools[[i]])
-  tools[[i]]<-ifelse(grepl(card12, toolChoices[[i]]) == TRUE, 12, tools[[i]])
-  }
-
-# Assign hypothesis and hypothesis level by round, based on tool choices
-
-round <- rep(list(1:nRounds), nElements)
-dummyList <- rep(list(rep(NA, nRounds)), nElements)
-
-# THE LOOPS and IF>ELSE STATEMENTS BELOW ARE RECURRENT AND COULD BE SIMPLIFIED WITH A FUNCTION
-# h2.2
-
-h2.2<-dummyList
-for (j in (1:nElements)) {
-  for (i in (1:nRounds)) {
-    if (grepl(card01, toolChoices[[j]][i]) == TRUE) {
-      h2.2[[j]][i]<-0
-      } else if (grepl(card02, toolChoices[[j]][i]) == TRUE) {
-        h2.2[[j]][i]<-1
-      }
-    }
-  }
-
-# h2.3
 
-h2.3<-dummyList
-for (j in 1:nElements) {
-  for (i in (1:nRounds)) {
-    if (grepl(card01, toolChoices[[j]][i]) == TRUE) {
-      h2.3[[j]][i]<-0
-    } else if (grepl(card03, toolChoices[[j]][i]) == TRUE) {
-      h2.3[[j]][i]<-1
-    }
-  }
+tool_dict <- list(
+    'TNTbarrel,SatchelCharge' = 1,
+    'BlackPowder,Dynamite' = 2,
+    'BlackPowder,RDX' = 3,
+    'RDX,Dynamite' = 4,
+    'Mine1,Mine2' = 5,
+    'Mine4,Mine3' = 6,
+    'Mine1,BlackPowder' = 7,
+    'Mine2,BlackPowder' = 7,
+    'Mine3,BlackPowder' = 8,
+    'Mine4,BlackPowder' = 8,
+    'TNTbarrel,Dynamite' = 9,
+    'BlackPowder,SatchelCharge' = 10,
+    'SatchelCharge,RDX' = 11,
+    'Dynamite,SatchelCharge' = 12
+)
+
+# define functions
+# helper function for grepping in h_values_complex
+get_search_value <- function(tools, value) {
+    return(paste(names(which(tools == value)), collapse = '|'))
 }
 
-# h2.4
-
-h2.4<-dummyList
-for (j in 1:nElements) {
-  for (i in (1:nRounds)) {
-    if (grepl(card04, toolChoices[[j]][i]) == TRUE) {
-      h2.4[[j]][i]<-0
-    } else if (grepl(card01, toolChoices[[j]][i]) == TRUE) {
-      h2.4[[j]][i]<-1
-    }
-  }
+# gets various h values, given input parameters for parsing
+h_values <- function(data, h, rds) {
+    tmp <- lapply(data, function(x) c(0:(length(h)-1))[sapply(h, grepl, x)])
+    return(mapply(function(x, n) {
+        rep(x, n)
+    }, tmp, rds, SIMPLIFY = FALSE))
 }
 
-# h2.5
-
-h2.5<-dummyList
-for (j in 1:nElements) {
-  for (i in (1:nRounds)) {
-    if (grepl(card05, toolChoices[[j]][i]) == TRUE) {
-      h2.5[[j]][i]<-0
-    } else if (grepl(card06, toolChoices[[j]][i]) == TRUE) {
-      h2.5[[j]][i]<-1
-    }
-  }
+# gets other h values that are complex, based on multiple conditions
+h_values_complex <- function(rounds, tool_choices, tool_set, v1, v2) {
+    return(mapply(function(rds, tools) {
+        tmp <- rep(NA, rds)
+        tmp[grep(get_search_value(tool_set, v1), tools)] <- 0
+        tmp[grep(get_search_value(tool_set, v2), tools)] <- 1
+        return(tmp)
+    }, rounds, tool_choices, SIMPLIFY = FALSE))
 }
 
+# Read in logs
+datalist <- lapply(list.files(dd, pattern = ".*.txt"), function(x) {

```

Comment:
I saved a "cookies.txt" file to the working folder, but now it's being picked up by this. We need the "pattern" to be more specific to the name of log files, which should be consistently something like "4_10_2018__15_20_44_0"